mne/chpi.py

```diff
@@ -976,14 +1067,26 @@ def compute_chpi_amplitudes(raw, t_step_min=0.01, t_window='auto',
                 % (len(hpi['freqs']), len(fit_idxs), tmax - tmin))
     del tmin, tmax
     sin_fits = dict()
+    sin_fits['proj'] = hpi['proj']
     sin_fits['times'] = np.round(fit_idxs + raw.first_samp -
                                  hpi['n_window'] / 2.) / raw.info['sfreq']
-    sin_fits['proj'] = hpi['proj']
-    sin_fits['slopes'] = np.empty(
-        (len(sin_fits['times']),
-         len(hpi['freqs']),
-         len(sin_fits['proj']['data']['col_names'])))
-    for mi, midpt in enumerate(ProgressBar(fit_idxs, mesg='cHPI amplitudes')):
+    n_times = len(sin_fits['times'])
+    n_freqs = len(hpi['freqs'])
+    n_chans = len(sin_fits['proj']['data']['col_names'])
+    if snr:
+        del sin_fits['proj']
+        sin_fits['freqs'] = hpi['freqs']
+        ch_types = raw.get_channel_types()
+        grad_offset = 3 if 'mag' in ch_types else 0
+        for ch_type in ('mag', 'grad'):
+            if ch_type in ch_types:
+                for key in ('snr', 'power', 'resid'):
+                    cols = 1 if key == 'resid' else n_freqs
+                    sin_fits[f'{ch_type}_{key}'] = np.empty((n_times, cols))
```

Comment:
This is the non-deterministic part. `np.empty` will not initialize to anything, so you just get whatever junk was in that block of memory that was allocated. When I change this to `np.zeros`, I get flat lines for all five runs:

![Screenshot from 2021-07-20 12-19-35](https://user-images.githubusercontent.com/2365790/126359863-9c7dfcc7-a120-47f9-b228-9268087d6a74.png)


So it looks like this never gets values assigned to it for grads for some reason (?).